src/SDKs/PolicyInsights/Management/PolicyInsights.Tests/PolicyInsightsTests.cs

```diff
@@ -0,0 +1,726 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Net;
+using Microsoft.Azure.Management.PolicyInsights;
+using Microsoft.Azure.Management.PolicyInsights.Models;
+using Microsoft.Azure.Test.HttpRecorder;
+using Microsoft.Rest.ClientRuntime.Azure.TestFramework;
+using PolicyInsights.Tests.Helpers;
+using Xunit;
+
+namespace PolicyInsights.Tests
+{
+    public class PolicyInsightsTests : TestBase
+    {
+        // Change this to Record when you want to record a test; also change the project to build for full .net since interactive login won't work for .net core
+        private static readonly HttpRecorderMode Mode = HttpRecorderMode.Playback;
+
+        #region Test setup
+
+        private static readonly string SubscriptionId = "d0610b27-9663-4c05-89f8-5b4be01e86a5";
+
+        static PolicyInsightsTests()
+        {
+            if (HttpRecorderMode.Record == Mode)
+            {
+                Environment.SetEnvironmentVariable("TEST_CSM_ORGID_AUTHENTICATION", "SubscriptionId=" + SubscriptionId + ";Environment=Prod;HttpRecorderMode=Record;");
+                Environment.SetEnvironmentVariable("AZURE_TEST_MODE", "Record");
+            }
+            else
+            {
+                Environment.SetEnvironmentVariable("TEST_CSM_ORGID_AUTHENTICATION", "SubscriptionId=" + SubscriptionId + ";Environment=Prod;HttpRecorderMode=Playback;");
+                Environment.SetEnvironmentVariable("AZURE_TEST_MODE", "Playback");
+            }
+
+            TestEnvironment = TestEnvironmentFactory.GetTestEnvironment();
+        }
+
+        public static TestEnvironment TestEnvironment { get; }
+
+        private PolicyInsightsClient GetPolicyInsightsClient(MockContext context)
+        {
+            if (HttpMockServer.Mode == HttpRecorderMode.Record)
+            {
+                HttpMockServer.RecordsDirectory = @"C:\Git\bulentelmaci\azure-sdk-for-net\src\SDKs\PolicyInsights\Management\PolicyInsights.Tests\SessionRecords";
```

Comment:
@bulentelmaci this will not work if I ever try to record your tests? Any reasons you are hardcoding these paths?

Reply:
I needed to use interactive login to record sessions which means having to use fill .net SDK. With full .net SDK, recording is broken unless you override recordsdirectory with a new one right after context is started because the recordsdirectory set is invalid (conflicts with the DLL name).